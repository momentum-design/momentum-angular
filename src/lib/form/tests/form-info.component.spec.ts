import { async, TestBed } from '@angular/core/testing';
import { FormInfoComponent } from '../form-info.component';

describe('FormInfoComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [FormInfoComponent],
    }).compileComponents();
  }));

  it('should create and match snapshot', () => {
    const fixture = TestBed.createComponent(FormInfoComponent);
    const testComponent = fixture.componentInstance;
    fixture.detectChanges();
    expect(testComponent).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render one FormInfo', () => {
    const fixture = TestBed.createComponent(FormInfoComponent);
    fixture.detectChanges();
    expect(fixture.nativeElement.className).toContain('section__info');
  });

  it('should render title', () => {
    const fixture = TestBed.createComponent(FormInfoComponent);
    const testComponent = fixture.componentInstance;
    testComponent.title = 'test';
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.section__title');
    expect(element.textContent).toEqual('test');
  });

  it('should render description', () => {
    const fixture = TestBed.createComponent(FormInfoComponent);
    const testComponent = fixture.componentInstance;
    testComponent.description = 'test';
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.section__description');
    expect(element.textContent).toEqual('test');
  });
});
