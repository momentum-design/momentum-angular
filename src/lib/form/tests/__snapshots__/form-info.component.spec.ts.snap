// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInfoComponent should create and match snapshot 1`] = `
<md-form-info
  description=""
  title=""
>
  <h4
    class="section__title"
  >
    
  </h4>
</md-form-info>
`;
